jboss-kie-smartrouter/added/launch/jboss-kie-smartrouter.sh

```diff
@@ -97,15 +92,53 @@ function configure_router_location {
     fi
     JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.kie.server.router.port=${kieServerRouterPort}"
 
+    local kieServerRouterProtocol="${KIE_SERVER_ROUTER_PROTOCOL}"
+    if [ "${kieServerRouterProtocol}" = "" ]; then
+        kieServerRouterProtocol="http"
+    fi
+
     local kieServerRouterUrlExternal="${KIE_SERVER_ROUTER_URL_EXTERNAL}"
+    local routeName="${KIE_SERVER_ROUTER_NAME// /_}"
+    local kieServerRouterAppName="${KIE_SERVER_ROUTER_APPLICATION_NAME}"
+    
+    # value from  KIE_SERVER_ROUTER_NAME (org.kie.server.router.name) if no route name is defined
+    if [ "${routeName}" = "KIE_Server_Router" ]; then
+        if [ "${kieServerRouterProtocol}" = "https" ]; then
+            routeName=secure-"${kieServerRouterAppName}"-smartrouter
+        else
+            routeName="${kieServerRouterAppName}"-smartrouter
+        fi
+    fi
+    
     if [ "${kieServerRouterUrlExternal}" = "" ]; then
-        local kieServerRouterProtocol="${KIE_SERVER_ROUTER_PROTOCOL}"
-        if [ "${kieServerRouterProtocol}" = "" ]; then
-            kieServerRouterProtocol="http"
+        # only execute the following lines if this container is running on OpenShift
+        if [ -e /var/run/secrets/kubernetes.io/serviceaccount/token ]; then
+            # try to retrieve the host from kubernetes api, a service account with "view" role is necessary to perform this request
+            local namespace=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
+            local token=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+            local response=$(curl -s -w "%{http_code}" --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
+                            -H "Authorization: Bearer $token" \
+                            -H 'Accept: application/json' \
+                            https://${KUBERNETES_SERVICE_HOST:-kubernetes.default.svc}:${KUBERNETES_SERVICE_PORT:-443}/apis/route.openshift.io/v1/namespaces/${namespace}/routes/${routeName})
         fi
-        kieServerRouterUrlExternal="${kieServerRouterProtocol}://${kieServerRouterHost}:${kieServerRouterPort}"
+
+        if [ "${response: -3}" = "200" ]; then
```

Comment:
same than above, don't replicate code, reuse this [function](https://github.com/jboss-container-images/jboss-kie-modules/blob/master/jboss-kie-wildfly-common/added/launch/jboss-kie-wildfly-common.sh#L64)

Reply:
same as above regarding re-use of function between wildfly and non-wildfly images.